### ISSUE TITLE
Please provide linux/arm64 Binary Release

### ISSUE THREAD

**Issue by naraesk** [realm/SwiftLint#4531]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug

Since v5.6 swift releases arm64 binaries (aarch64) for Linux/Ubuntu. The swiftlint binary for Linux only runs on amd64. For arm, I have to build swiftlint myself. This is of course possible, but it would be great if arm64 binaries would be provided. This would simplify the integration into arm-based docker containers and Github Actions.

### Environment

* SwiftLint version (run `swiftlint version` to be sure)? 5.7.1
* Installation method used (Homebrew, CocoaPods, building from source, etc)? building from source

**Comment by jpsim** [comment]
Yes this would be great, and we're already using Docker Buildx so in theory it should be possible to update our CI to run that.

I tried a few months ago but the Swift 5.6.x Docker images didn't come in arm64 flavours at the time. Now I think the Swift 5.7 images do have an arm64 variant. I'll give this another shot.

**Comment by naraesk** [comment]
Great, thanks for already working on this. :-)

To clarify, I'm primarily interested in an arm64 binary published on https://github.com/realm/SwiftLint/releases which I can integrate into my own docker container.

**Comment by doozMen** [comment]
A grate blog post can be found https://www.polpiella.dev/binary-targets-in-modern-swift-packages

**Comment by jpsim** [comment]
The Linux binaries in the release assets are generated in Docker, so that’s the first step.

Unfortunately it looks like with QEMU and the default GitHub Actions machines they’re not beefy enough to finish building both architectures, and the CI job appears to be getting killed.

**Comment by doozMen** [comment]
Yes I have similar issues, ... shitty

**Comment by doozMen** [comment]
I managed to fix it by manually building an artifactbundle as described here https://github.com/airbnb/swift/discussions/197#discussioncomment-4109195

Would it be possible for swiftlint to provide this in the future?

**Comment by inket** [comment]
Sorry to hijack/bump. I noticed newer versions also don't have the Linux x86 binary. (We use SwiftLint in GitHub actions with the Linux runner as it's a lot cheaper than macOS runners.)

Is there anything I can help with?

If it's too complicated for the foreseeable future, may I suggest building the Linux versions in a separate GitHub action?

**Comment by jpsim** [comment]
> Sorry to hijack/bump. I noticed newer versions also don't have the Linux x86 binary. (We use SwiftLint in GitHub actions with the Linux runner as it's a lot cheaper than macOS runners.)

I missed a manual step in the 0.50.1, the x86 Linux binary is up now: https://github.com/realm/SwiftLint/releases/tag/0.50.1

Direct link: https://github.com/realm/SwiftLint/releases/download/0.50.1/swiftlint_linux.zip

**Comment by jagreenwood** [comment]
Hi there, I was wondering there were any updates to this issue worth noting. I too have a need for a linux/arm64 binary release.

**Comment by SimplyDanny** [comment]
> Hi there, I was wondering there were any updates to this issue worth noting. I too have a need for a linux/arm64 binary release.

There's work going on already. This needed dependencies to be updated first.

**Comment by jagreenwood** [comment]
> There's work going on already. This needed dependencies to be updated first.

Great thanks for the update! This would be a huge time-saver when running SwiftLint in devcontainers, GitHub actions, and beyond :)

**Comment by gmazzo** [comment]
Any update on `linux/arm64` binaries/docker containers? We are currently migrating to ARM GHA runners, and this is becoming a blocker. I tried building from scratch but having tons of issues 😅

**Comment by SimplyDanny** [comment]
If everything goes well, we'll have an arm64 Docker image as well as an arm64 binary available with the next release.
